src/Juvix/Compiler/Core/Transformation/Check/Base.hs

```diff
@@ -1,22 +1,36 @@
 module Juvix.Compiler.Core.Transformation.Check.Base where
 
 import Juvix.Compiler.Core.Data.InfoTable
+import Juvix.Compiler.Core.Data.InfoTableBuilder
 import Juvix.Compiler.Core.Data.TypeDependencyInfo (createTypeDependencyInfo)
 import Juvix.Compiler.Core.Error
 import Juvix.Compiler.Core.Extra
-import Juvix.Compiler.Core.Info.LocationInfo (getInfoLocation)
+import Juvix.Compiler.Core.Info.LocationInfo (getInfoLocation, getNodeLocation)
 import Juvix.Compiler.Core.Info.TypeInfo qualified as Info
 import Juvix.Compiler.Core.Language
+import Juvix.Compiler.Core.Transformation.Base (mapT')
+import Juvix.Data.NameKind
 import Juvix.Data.PPOutput
 
 dynamicTypeError :: Node -> Maybe Location -> CoreError
 dynamicTypeError node loc =
   CoreError
-    { _coreErrorMsg = ppOutput $ "compilation for this target requires full type information",
+    { _coreErrorMsg = ppOutput "compilation for this target requires full type information",
       _coreErrorNode = Just node,
       _coreErrorLoc = fromMaybe defaultLoc loc
     }
 
+axiomError :: Members '[Error CoreError, InfoTableBuilder] r => Symbol -> Maybe Location -> Sem r a
+axiomError sym loc = do
+  tbl <- getInfoTable
+  let nameTxt = identName tbl sym
+  throw
+    CoreError
+      { _coreErrorMsg = ppOutput ("The symbol" <+> annotate (AnnKind KNameAxiom) (pretty nameTxt) <> " is defined as an axiom and thus it cannot be compiled"),
```

Comment:
Isn't some function for `annotate (AnnKind KNameAxiom)..`? If not, let us define them,
e.g. `annAxiom` and similarly for other kinds.

Reply:
I don't think we need an extra function for each Kind annotation. But, if we really want them, we should refactor that in a separate PR, as this would entail a lot of changes not related to this

Reply:
Sure, I didn't intend to include those changes here.